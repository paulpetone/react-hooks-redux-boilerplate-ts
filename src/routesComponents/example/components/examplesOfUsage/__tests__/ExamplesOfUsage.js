import ExamplesOfUsage from '../'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { render, fireEvent } from '@testing-library/react'
import * as Gatsby from 'gatsby'

const mockStore = configureStore([])

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

useStaticQuery.mockImplementation(() => ({
  example: {
    nameWithOwner: 'gatsbyjs/gatsby',
    url: 'https://github.com/gatsbyjs/gatsby',
    anyOtherData: {
      hi: 'hola',
      owner: {
        id: 12551863,
      },
    },
  },
}))

describe('ExamplesOfUsage', () => {
  let component
  let mockedStore

  beforeEach(() => {
    mockedStore = mockStore({
      items: ['store item'],
    })

    component = render(
      <Provider store={mockedStore}>
        <ExamplesOfUsage />
      </Provider>
    )
  })

  it('renders correctly', () => {
    expect(component.baseElement).toMatchSnapshot()
  })

  it('renders example 1 with add item feature', () => {
    mockedStore.dispatch = jest.fn()
    const { getByText, getByLabelText } = component
    const button = getByText('Add Item')
    const input = getByLabelText('Add Item:')

    fireEvent.change(input, { target: { value: 'new item' } })
    fireEvent.click(button, new MouseEvent('click'))

    expect(mockedStore.dispatch).toHaveBeenCalledTimes(1)
  })

  it('renders example 3 with gatsby data', () => {
    const { getByText } = component

    getByText('Url: https://github.com/gatsbyjs/gatsby')
    getByText('NameWithOwner: gatsbyjs/gatsby')
    getByText('AnyOtherData.Hi: hola')
    getByText('AnyOtherData.Owner.id: 12551863')
  })
})
