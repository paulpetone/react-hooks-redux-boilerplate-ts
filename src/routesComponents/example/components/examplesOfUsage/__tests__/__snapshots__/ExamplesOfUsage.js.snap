// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExamplesOfUsage renders correctly 1`] = `
<body>
  <div>
    <div
      class="examples"
    >
      This is an example using all u gonna need to create any component (store, handlers, state, effects and graphql)
      <br />
      <b>
        Take a look at @routesComponents/example/examplesOfUsage component
      </b>
      <hr />
      <h2>
        Example 1
      </h2>
      <i>
        This first 
        &lt;ul&gt;
         uses a simple reducer
      </i>
      <ul>
        <li>
          store item
        </li>
      </ul>
      <form>
        <label
          for="example-1"
        >
          Add Item:
        </label>
        <input
          id="example-1"
          type="text"
        />
        <button
          type="submit"
        >
          Add Item
        </button>
      </form>
      <hr />
      <h2>
        Example 2
      </h2>
      <i>
        This second 
        &lt;ul&gt;
         uses an async information from store
      </i>
      <ul />
      <hr />
      <h2>
        Example 3
      </h2>
      <i>
        This third example uses gatsby-node.js to fetch data from API and set it to Graphql, and to retrieve this data we are using the following file: @routesComponents/example/examplesOfUsage/hooks/useGraphQLData.tsx 
        <br />
        This is very useful for SEO issues!
      </i>
      <br />
      <br />
      <i>
        Url: 
        https://github.com/gatsbyjs/gatsby
      </i>
      <br />
      <br />
      <i>
        NameWithOwner: 
        gatsbyjs/gatsby
      </i>
      <br />
      <br />
      <i>
        AnyOtherData.Hi: 
        hola
      </i>
      <br />
      <br />
      <i>
        AnyOtherData.Owner.id: 
        12551863
      </i>
    </div>
  </div>
</body>
`;
